models/quality_measures/intermediate/nqf0034_colorectal_cancer_screening/quality_measures__int_nqf0034_exclude_advanced_illness.sql

```diff
@@ -0,0 +1,323 @@
+            , med_claim_exclusions.claim_end_date
+          ) as exclusion_date
+        , med_claim_exclusions.concept_name
+            || ' with '
+            || condition_exclusions.concept_name
+          as exclusion_reason
+        , med_claim_exclusions.concept_category
+        , med_claim_exclusions.qualifying_count
+    from patients_with_frailty
+         inner join med_claim_exclusions
+            on patients_with_frailty.patient_id = med_claim_exclusions.patient_id
+         inner join condition_exclusions
+            on med_claim_exclusions.claim_id = condition_exclusions.claim_id
+
+
+    union all
+
+    select distinct
+          patients_with_frailty.patient_id
+        , observation_exclusions.observation_date as exclusion_date
+        , observation_exclusions.concept_name as exclusion_reason
+        , observation_exclusions.concept_category
+        , observation_exclusions.qualifying_count
+    from patients_with_frailty
+         inner join observation_exclusions
+            on patients_with_frailty.patient_id = observation_exclusions.patient_id
+         inner join condition_exclusions
+             on observation_exclusions.patient_id = condition_exclusions.patient_id
+             and observation_exclusions.observation_date = condition_exclusions.recorded_date
+
+    union all
+
+    select distinct
+          patients_with_frailty.patient_id
+        , procedure_exclusions.procedure_date as exclusion_date
+        , procedure_exclusions.concept_name as exclusion_reason
+        , procedure_exclusions.concept_category
+        , procedure_exclusions.qualifying_count
+    from patients_with_frailty
+         inner join procedure_exclusions
+         on patients_with_frailty.patient_id = procedure_exclusions.patient_id
+         inner join condition_exclusions
+         on procedure_exclusions.patient_id = condition_exclusions.patient_id
+         and procedure_exclusions.procedure_date = condition_exclusions.recorded_date
+
+
+)
+
+/*
+    Patients greater than or equal to 66 with at least one claim/encounter for
+    frailty during the measurement period
+
+    AND two outpatient, observation, ED or nonacute inpatient encounters
+    on different dates of service with an advanced illness diagnosis during
+    measurement period or the year prior to measurement period
+*/
+
+/*
+    Filter to patients who have had one acute inpatient encounter or
+    two nonacute outpatient encounters
+*/
+
+
+, qualifying_encounters as (
+
+    select
+      patient_id
+    , exclusion_date
+    , exclusion_reason
+from encounters_with_conditions e
+qualify dense_rank() over(partition by patient_id,concept_category order by exclusion_date) >= qualifying_count
+)
+
+select
+      patient_id
+    , exclusion_date
+    , exclusion_reason
+    , '{{ var('tuva_last_run')}}' as tuva_last_run
+from qualifying_encounters
+-- qualify row_number() over (partition by patient_id order by exclusion_date) = 1
```

Comment:
@tuvaforrest Should this be commented out?

Reply:
Ok, whatever you want. IMO, we shouldn't leave unused code in the project. I think it's confusing.